react/ProductAssemblyOptions.tsx

```diff
@@ -0,0 +1,142 @@
+import React, { useMemo, FC, Fragment } from 'react'
+import useProduct from 'vtex.product-context/useProduct'
+import { find, propEq, compose, last, split, pathOr } from 'ramda'
+import ProductAssemblyOptionsGroupStateManager from './components/ProductAssemblyOptions/ProductAssemblyOptionsGroupStateManager'
+
+type PriceMap = Record<string, Record<string, Record<string, number>>>
+
+const findItemMetadata = (id: string) => find<MetadataItem>(propEq('id', id))
+const splitGroupName = compose(
+  last,
+  split('_')
+)
+
+const getItemAssemblyOptions = (
+  selectedItem: SelectedItem,
+  itemMetadata: ItemMetadata
+) => {
+  if (!itemMetadata || !itemMetadata.items) {
+    return null
+  }
+  const metadata = findItemMetadata(selectedItem.itemId)(itemMetadata.items)
+  return metadata && metadata.assemblyOptions
+}
+
+const parsePriceMap = (itemMetadata: ItemMetadata) => {
+  const result = {} as PriceMap
+  for (const priceTableItem of itemMetadata.priceTable) {
+    const { type } = priceTableItem
+    result[type] = {}
+    for (const priceItem of priceTableItem.values) {
+      const groupPrices = result[type][priceItem.assemblyId] || {}
+      groupPrices[priceItem.id] = priceItem.price
+      result[type][priceItem.assemblyId] = groupPrices
+    }
+  }
+  return result
+}
+
+const parseAssemblyOptions = (
+  assemblyOptions: AssemblyOption[],
+  priceMap: PriceMap,
+  product: Product,
+  fatherItemId?: string,
+  treePath?: TreePath[]
+) => {
+  const assemblyOptionsParsed = {} as Record<string, AssemblyOptionGroup>
+  for (const assemblyOption of assemblyOptions) {
+    if (!assemblyOption.composition) {
```

Comment:
Why are you checking on the treepath and composition?

Reply:
composition may be null in some groups. Those with a null value are not interesting for us.